### PR TITLE
SHS-5958: Bug: Inconsistent "Add Above" feature

### PR DESCRIPTION
**Pull request by codechefmarc**
# READY FOR REVIEW

## Summary
- Adds "Add between" buttons to paragraphs on Flexible Pages and Private Pages.

## Need Review By (Date)
2024-12-04

## Urgency
medium

## Steps to Test
1. Login to the site
2. Add a new Flexible Page or Private Page
3. Verify that there are "+ Add" buttons for paragraphs

Screenshot 2024-12-04 at 11 42 51 AM

5. Add a paragraph
6. Verify that there is also an "+Add" button that gets added when hovering over a paragraph to add in between
7. Edit an existing Flexible Page or Private Page
8. Verify that the "+Add" buttons show up there as well

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208992447859410

### CODE REVIEW

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by codechefmarc** [comment]
@cienvaras - This is ready for review. Note - I had to update some of the tests to click on the new add in-between buttons so those worked. Though I'm still getting some of those reoccurring test errors for unrelated things. I think I fixed the `testPhotoAlbum` one - it was because the photo album could be above or below the text area and sometimes it would fail it it was above it as the test was specifically calling item 1 (second paragraph). I changed it to use an XPath selector to look specifically for `stanford_gallery` but that test didn't run in the last test run so not sure. I'm out of hours for this week, so I can continue next week on these tests if needed. Thanks!

File: config/default/core.entity_form_display.node.hs_basic_page.default.yml [hunk 1]
```diff
@@ -79,22 +83,31 @@ content:
...
           dialog_off_canvas: false
           dialog_style: tiles
+          drag_drop: false
   field_menulink:
     type: menu_link_default
     weight: 26
     region: content
     settings: {  }
     third_party_settings: {  }
+  layout_builder__layout:
```
Comment (cienvaras) [inline config/default/core.entity_form_display.node.hs_basic_page.default.yml]: Not related to the current task, please revert.

File: config/default/core.entity_form_display.node.hs_private_page.default.yml [hunk 2]
```diff
@@ -78,6 +83,12 @@ content:
           audio: 0
           video: 0
         default: link
+  layout_builder__layout:
```
Comment (cienvaras) [inline config/default/core.entity_form_display.node.hs_private_page.default.yml]: Not related to the current task, please revert.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Review by cienvaras — CHANGES_REQUESTED** [review]
@codechefmarc Thanks, looks great. There are just a couple of things to fix in the exported config.

I think I have an idea of why the tests are failing. I'll try a couple of fixes and let you know how it goes.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

File: config/default/core.entity_form_display.node.hs_basic_page.default.yml (hunk 1 above)
Comment (codechefmarc) [reply config/default/core.entity_form_display.node.hs_basic_page.default.yml]: Removed.

File: config/default/core.entity_form_display.node.hs_private_page.default.yml (hunk 2 above)
Comment (codechefmarc) [reply config/default/core.entity_form_display.node.hs_private_page.default.yml]: Removed.

File: config/default/core.entity_form_display.node.hs_basic_page.default.yml [hunk 3]
```diff
@@ -145,5 +152,4 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
-hidden:
-  layout_builder__layout: true
+hidden: {  }
```
Comment (cienvaras) [inline config/default/core.entity_form_display.node.hs_basic_page.default.yml]: @codechefmarc  Sorry, I didn't highlighted this in the review, but this needs to be reverted too, it's related to the other change you reverted before.

File: config/default/core.entity_form_display.node.hs_private_page.default.yml [hunk 4]
```diff
@@ -128,5 +133,4 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
-hidden:
-  layout_builder__layout: true
+hidden: {  }
```
Comment (cienvaras) [inline config/default/core.entity_form_display.node.hs_private_page.default.yml]: @codechefmarc See previous comment.

**Review by cienvaras — CHANGES_REQUESTED** [review]
@codechefmarc We missed a couple of changes that need to be reverted, please see inline comments. Thanks!

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1692-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-colorful-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://hs-traditional-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://english-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://history-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com
- https://west-tarbt8wa9abxmb9umqmivtvhkizud7vq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460

Visual Diffs:
- https://dashboard.tugboatqa.com/674faf60dab10e2ceed5b460/visualdiffs

File: config/default/core.entity_form_display.node.hs_basic_page.default.yml (hunk 3 above)
Comment (codechefmarc) [reply config/default/core.entity_form_display.node.hs_basic_page.default.yml]: Reverted.

File: config/default/core.entity_form_display.node.hs_private_page.default.yml (hunk 4 above)
Comment (codechefmarc) [reply config/default/core.entity_form_display.node.hs_private_page.default.yml]: Reverted.

**Review by cienvaras — APPROVED** [review]
LGTM!

@ahughes3 This is ready for you to review.

**Review by ahughes3 — APPROVED** [review]
Looks good!

**Review by pookmish — APPROVED** [review]

**Pull request merged**